openwisp_users/admin.py

```diff
@@ -510,6 +512,67 @@ def has_delete_permission(self, request, obj=None):
             return False
         return super().has_delete_permission(request, obj)
 
+    def delete_view(self, request, object_id, extra_context=None):
+        try:
+            return super().delete_view(request, object_id, extra_context)
+        except OwnershipRequired:
+            self.message_user(
+                request,
+                _(
+                    "Can't delete this organization user because "
+                    'it belongs to an organization owner'
+                ),
+                messages.ERROR,
+            )
+            redirect_url = reverse(
+                f'admin:{self.model._meta.app_label}_organizationuser_change',
+                args=[object_id],
+            )
+            return HttpResponseRedirect(redirect_url)
+
+    def get_actions(self, request):
+        actions = super().get_actions(request)
+        if not request.POST.get('post') and 'delete_selected' in actions:
+            del actions['delete_selected']
+        return actions
+
+    def delete_selected_overridden(self, request, queryset):
+        count = 0
+        pks = []
+        for obj in queryset:
+            if obj.user.is_owner(obj.organization_id):
```

Comment:
please add an `assertNumQueries` on the tests which trigger this line: to me it looks it could generate too many queries.
If it does generate too many queries, `select_related` has to be used.

Reply:
@nemesisdesign i don't think it is in this PR that we should be checking if the `helpers` produce extra queries or not. I believe that should have been done in the PR which had that solution. But from the comments below, it shows no extra queries are executed here.

Reply:
the test does not have the aim of testing the helper, we already have a test for that, it has the aim to test this method, which may be changed by somebody else in the future and hence may break it or modify in a way that generates too many queries, so please do add it, as I already explained we need to start to keep in check the number of queries our code generates because it's getting out of hand. We already started this process: https://github.com/search?q=org%3Aopenwisp+assertNumQueries&type=Code
We have to stick at it and that's why I'm insisting with you. Please make it a habit, it's a very good one.